packages/components/src/components/alert/alert.tsx

```diff
@@ -25,11 +33,50 @@ export class alert {
     );
   }
 
+  onCloseAlert = () => {
+    this.opened = false;
+  };
+
+  @Method()
+  open() {
+    this.opened = true;
+  }
+
+  onCloseAlertWithTimeout = () => {
+    if (this.timeout !== undefined) {
+      setTimeout(this.onCloseAlert, this.timeout);
+    } else if (!this.withTimeout) {
+      let defaultTimeout = 500;
```

Comment:
As the default timeout is set directly on the prop (prev comment), you can check if `withTimeout === true` and only then use the `timeout` prop